test/Furlongs.jl

```diff
@@ -0,0 +1,114 @@
+Base.eps(::Type{Furlong{p,T}}) where {p,T<:AbstractFloat} = Furlong{p}(eps(T))
+Base.eps(::Furlong{p,T}) where {p,T<:AbstractFloat} = eps(Furlong{p,T})
+Base.floatmin(::Type{Furlong{p,T}}) where {p,T<:AbstractFloat} = Furlong{p}(floatmin(T))
+Base.floatmin(::Furlong{p,T}) where {p,T<:AbstractFloat} = floatmin(Furlong{p,T})
+Base.floatmax(::Type{Furlong{p,T}}) where {p,T<:AbstractFloat} = Furlong{p}(floatmax(T))
+Base.floatmax(::Furlong{p,T}) where {p,T<:AbstractFloat} = floatmax(Furlong{p,T})
+Base.conj(x::Furlong{p,T}) where {p,T} = Furlong{p,T}(conj(x.val))
+
+# convert Furlong exponent p to a canonical form
+canonical_p(p) = isinteger(p) ? Int(p) : Rational{Int}(p)
+
+Base.abs(x::Furlong{p}) where {p} = Furlong{p}(abs(x.val))
+Base.abs2(x::Furlong{p}) where {p} = Furlong{canonical_p(2p)}(abs2(x.val))
+Base.inv(x::Furlong{p}) where {p} = Furlong{canonical_p(-p)}(inv(x.val))
+
+for f in (:isfinite, :isnan, :isreal, :isinf)
+    @eval Base.$f(x::Furlong) = $f(x.val)
+end
+for f in (:real,:imag,:complex,:+,:-)
+    @eval Base.$f(x::Furlong{p}) where {p} = Furlong{p}($f(x.val))
+end
+
+import Base: +, -, ==, !=, <, <=, isless, isequal, *, /, //, div, rem, mod, ^
+for op in (:+, :-)
+    @eval function $op(x::Furlong{p}, y::Furlong{p}) where {p}
+        v = $op(x.val, y.val)
+        Furlong{p}(v)
+    end
+end
+for op in (:(==), :(!=), :<, :<=, :isless, :isequal)
+    @eval $op(x::Furlong{p}, y::Furlong{p}) where {p} = $op(x.val, y.val)::Bool
+end
+for (f,op) in ((:_plus,:+),(:_minus,:-),(:_times,:*),(:_div,://))
+    @eval function $f(v::T, ::Furlong{p}, ::Union{Furlong{q},Val{q}}) where {T,p,q}
+        s = $op(p, q)
+        Furlong{canonical_p(s),T}(v)
+    end
+end
+for (op,eop) in ((:*, :_plus), (:/, :_minus), (://, :_minus), (:div, :_minus))
+    @eval begin
+        $op(x::Furlong{p}, y::Furlong{q}) where {p,q} =
+            $eop($op(x.val, y.val),x,y)
+        $op(x::Furlong{p}, y::S) where {p,S<:TNumber} = $op(x,Furlong{0,S}(y))
+        $op(x::S, y::Furlong{p}) where {p,S<:TNumber} = $op(Furlong{0,S}(x),y)
+    end
+end
+# to fix an ambiguity
+//(x::Furlong, y::Complex) = x // Furlong{0,typeof(y)}(y)
+for op in (:rem, :mod)
+    @eval begin
+        $op(x::Furlong{p}, y::Furlong) where {p} = Furlong{p}($op(x.val, y.val))
+        $op(x::Furlong{p}, y::TNumber) where {p} = Furlong{p}($op(x.val, y))
+    end
+end
+Base.sqrt(x::Furlong) = _div(sqrt(x.val), x, Val(2))
+Base.muladd(x::Furlong, y::Furlong, z::Furlong) = x*y + z
+Base.muladd(x::Furlong, y::TNumber, z::TNumber) = x*y + z
+Base.muladd(x::Furlong, y::Furlong, z::TNumber) = x*y + z
+Base.muladd(x::TNumber, y::Furlong, z::TNumber) = x*y + z
+Base.muladd(x::TNumber, y::TNumber, z::Furlong) = x*y + z
+Base.muladd(x::TNumber, y::Furlong, z::Furlong) = x*y + z
+Base.muladd(x::Furlong, y::TNumber, z::Furlong) = x*y + z
+
+Base.exp(f::Furlongs.Furlong) = exp(f.val)
+Base.cos(f::Furlongs.Furlong) = cos(f.val)
+Base.sin(f::Furlongs.Furlong) = sin(f.val)
```

Comment:
These still don't seem reasonable to me - IMO they only exist for dimensionless numbers (ie Furlong{0}). This is also how they're defined in Unitful IIRC.

Reply:
Cf. https://github.com/JuliaDiff/ForwardDiff.jl/pull/758#discussion_r2365945037

The test currently entails `exp` etc. being defined for `
`::Furlongs.Furlong{2, Float64}`:
https://github.com/JuliaDiff/ForwardDiff.jl/actions/runs/17889140228/job/50867233017#step:6:156

If my comment makes no sense, please elaborate on what is unreasonable - I'm quite sure I may not have understood what you mean.